frontend/packages/metal3-plugin/src/components/add-host.tsx

```diff
@@ -63,16 +63,49 @@ const buildBaremetalHostObject = (name, namespace, controller, online) => ({
   },
 });
 
-const AddHost: React.FC = () => {
+const AddHost: React.FC = ({ match }) => {
   const title = 'Create Bare Metal Host';
 
   const [formValue, setFormValue] = React.useState(initFormFields());
   const [inProgress, setProgress] = React.useState(false);
   const [error, setError] = React.useState('');
+  const [baremetalhost, setBaremetalhost] = React.useState({});
+  const [isFetching, setIsFetching] = React.useState(false);
+  const [hasBaremetalhost, setHasBaremetalhost] = React.useState(false);
+
+  // Get the name from the URL (if there is one)
+  const { params: { name: nameFromUrl } = {} } = match;
+  // If the name is set in the URL we'd like to set it in the form value
+  // and retrieve the object from the API.
+  if (nameFromUrl) {
```

Comment:
This is a side effect and needs to be wrapped in useEffect hook. We don't want to do it on every render